vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/AbstractTestRunResultListener.kt

```diff
@@ -1,48 +1,70 @@
 package com.malinskiy.marathon.android.executor.listeners
 
 import com.android.ddmlib.testrunner.TestIdentifier
+import com.malinskiy.marathon.execution.TestBatchResults
 import com.android.ddmlib.testrunner.TestRunResult as DdmLibTestRunResult
 
 abstract class AbstractTestRunResultListener : NoOpTestRunListener() {
 
-    private val runResult: DdmLibTestRunResult = DdmLibTestRunResult()
+    val runResult: DdmLibTestRunResult = DdmLibTestRunResult()
+
+    var active = true
 
     override fun testRunStarted(runName: String, testCount: Int) {
-        runResult.testRunStarted(runName, testCount)
+        synchronized(runResult) {
```

Comment:
this class should not be used from many threads and synchronization is not required here